liota/core/package_manager.py

```diff
@@ -448,9 +461,21 @@ def _package_chk_exists(self, file_name, ext_forced=None):
         if package_path.endswith(c_slash):
             c_slash = ""
         path_file = os.path.abspath(package_path + c_slash + file_name)
+        if not (path_file.startswith(os.path.abspath(package_path)+'/')):
+            log.error("Package %s is NOT under package path %s"
+                         % (file_name, package_path))
+            return None, None
 
         file_ext = None
         extensions = ["py", "pyc", "pyo"]
+
+        # get checksum for all candidates: path_file + ".py/.pyc/.pyo"
+        checksum_list = []
+        for ext in extensions:
```

Comment:
I had ever thought of this issue.
path_file_ext could be .py /.pyc or .pyo, which is decided by package manager. The user only specify the Package Name without extension. Therefor, we need to use checksum_list.